### PR TITLE
 JAMES-2729 RRT routes check domain exist

### PR DESCRIPTION
**Pull request by trantienduchn**

### CODE REVIEW

File: server/data/data-api/src/test/java/org/apache/james/rrt/lib/MappingSourceTest.java [hunk 1]
```diff
@@ -49,14 +50,14 @@ void asMailAddressStringShouldSerializeWilcard() {
 
     @Test
     void asMailAddressStringShouldSerializeDomain() {
-        MappingSource mappingSource = MappingSource.fromDomain(Domain.of(DOMAIN));
+        MappingSource mappingSource = MappingSource.fromDomain(Domain.of(DOMAIN_AS_STRING));
```
Comment (Arsnael) [inline server/data/data-api/src/test/java/org/apache/james/rrt/lib/MappingSourceTest.java]: Can use directly `DOMAIN` here?

**Comment by trantienduchn** [comment]
```
[4a78eea8a1b617bfd6c38c8b4be53cab513d1055] [ERROR] org.apache.james.rrt.memory.RewriteTablesTest.address mapping should be retrieved when searching with a domain alias
[4a78eea8a1b617bfd6c38c8b4be53cab513d1055] [ERROR]   Run 1: RewriteTablesTest » NullPointer
[4a78eea8a1b617bfd6c38c8b4be53cab513d1055] [ERROR]   Run 2: RewriteTablesTest » NullPointer
```

**Comment by trantienduchn** [comment]
test this please

**Comment by trantienduchn** [comment]
```
[494611adb4ad9962e0b24721073a69c5b8d91b35] [ERROR] Errors: 
[494611adb4ad9962e0b24721073a69c5b8d91b35] [ERROR]   IsSenderInRRTLoopTest.matchShouldReturnEmptyWhenLoopButNoRecipient:98 » NullPointer
[494611adb4ad9962e0b24721073a69c5b8d91b35] [ERROR]   IsSenderInRRTLoopTest.matchShouldReturnEmptyWhenNoRRTLoop:71 » NullPointer
[494611adb4ad9962e0b24721073a69c5b8d91b35] [ERROR]   IsSenderInRRTLoopTest.matchShouldReturnRecipientsWhenLoop:84 » NullPointer
```

**Comment by trantienduchn** [comment]
test this please

**Comment by Arsnael** [comment]
```
[baffd73a1ade92e853ddccac5599ad62128e9482] [ERROR] Errors: 
[baffd73a1ade92e853ddccac5599ad62128e9482] [ERROR]   SmtpStarttlsCommandTest.shouldNotRejectContentStartsWithStartTls » NoClassDefFound
[baffd73a1ade92e853ddccac5599ad62128e9482] [ERROR]   SmtpStarttlsCommandTest.shouldNotRejectContentWithStartTls » ExceptionInInitializer
[baffd73a1ade92e853ddccac5599ad62128e9482] [ERROR]   SmtpStarttlsCommandTest.shouldNotRejectRcptWithStartTls » NoClassDefFound Coul...
[baffd73a1ade92e853ddccac5599ad62128e9482] [ERROR]   SmtpStarttlsCommandTest.starttlsShouldBeRejectedWhenFollowedByCommand » NoClassDefFound
[baffd73a1ade92e853ddccac5599ad62128e9482] [ERROR]   SmtpStarttlsCommandTest.starttlsShouldWork » NoClassDefFound Could not initial...
```

**Comment by Arsnael** [comment]
test this please

**Comment by trantienduchn** [comment]
test this please

File: server/data/data-library/src/test/java/org/apache/james/rrt/lib/RecipientRewriteTableFixutre.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.rrt.lib;
+
+import org.apache.james.core.Domain;
+import org.apache.james.domainlist.api.DomainListException;
+import org.apache.james.domainlist.api.mock.SimpleDomainList;
+
+public interface RecipientRewriteTableFixutre {
```
Comment (Arsnael) [inline server/data/data-library/src/test/java/org/apache/james/rrt/lib/RecipientRewriteTableFixutre.java]: s/Fixutre/Fixture

File: server/data/data-library/src/main/java/org/apache/james/rrt/lib/AbstractRecipientRewriteTable.java [hunk 3]
```diff
@@ -184,6 +187,7 @@ public void addRegexMapping(MappingSource source, String regex) throws Recipient
 
         Mapping mapping = Mapping.regex(regex);
         checkDuplicateMapping(source, mapping);
+        checkDomainMappingSourceIsManaged(source);
```
[the remark this answers is not in the document]
Comment (trantienduchn) [reply server/data/data-library/src/main/java/org/apache/james/rrt/lib/AbstractRecipientRewriteTable.java]: Yes, I think

File: server/data/data-library/src/main/java/org/apache/james/rrt/lib/AbstractRecipientRewriteTable.java [hunk 4]
```diff
@@ -234,6 +239,8 @@ public void addErrorMapping(MappingSource source, String error) throws Recipient
         Mapping mapping = Mapping.error(error);
 
         checkDuplicateMapping(source, mapping);
+        checkDomainMappingSourceIsManaged(source);
```
[the remark this answers is not in the document]
Comment (trantienduchn) [reply server/data/data-library/src/main/java/org/apache/james/rrt/lib/AbstractRecipientRewriteTable.java]: idem

File: server/data/data-library/src/main/java/org/apache/james/rrt/lib/AbstractRecipientRewriteTable.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (trantienduchn) [reply server/data/data-library/src/main/java/org/apache/james/rrt/lib/AbstractRecipientRewriteTable.java]: adding regex mapping can be performed through JMX command, which is still need validation for user inputs.

**Comment by trantienduchn** [comment]
```
[c3b8b5ddf31744b4fc76f8bbe439496a29802893] [ERROR] Errors: 
[c3b8b5ddf31744b4fc76f8bbe439496a29802893] [ERROR]   DataCommandsIntegrationTest.addAddressMappingShouldWork » SourceDomainIsNotInDomainList
[c3b8b5ddf31744b4fc76f8bbe439496a29802893] [ERROR]   DataCommandsIntegrationTest.addRegexMappingShouldWork » SourceDomainIsNotInDomainList
[c3b8b5ddf31744b4fc76f8bbe439496a29802893] [ERROR]   DataCommandsIntegrationTest.listMappingsShouldWork » SourceDomainIsNotInDomainList
[c3b8b5ddf31744b4fc76f8bbe439496a29802893] [ERROR]   DataCommandsIntegrationTest.listUsersDomainMappingShouldWork » SourceDomainIsNotInDomainList
[c3b8b5ddf31744b4fc76f8bbe439496a29802893] [ERROR]   DataCommandsIntegrationTest.removeAddressMappingShouldWork » SourceDomainIsNotInDomainList
[c3b8b5ddf31744b4fc76f8bbe439496a29802893] [ERROR]   DataCommandsIntegrationTest.removeRegexMappingShouldWork » SourceDomainIsNotInDomainList
```

**Comment by trantienduchn** [comment]
```
[0cbd0d56e589d1f1d4c20dca1f3345d92dad5491] [ERROR]   WebAdminServerIntegrationTest.cassandraMappingsEndpointShouldKeepDataConsistencyWhenDataValid:413 1 expectation failed.
[0cbd0d56e589d1f1d4c20dca1f3345d92dad5491] JSON path source doesn't match.
[0cbd0d56e589d1f1d4c20dca1f3345d92dad5491] Expected: (a collection containing "alias1@domain.com" and a collection containing "alias2@domain.com")
[0cbd0d56e589d1f1d4c20dca1f3345d92dad5491]   Actual: []
```

**Comment by trantienduchn** [comment]
```
[ae075ee89251cdf1536d0b3387c9c8f1ca750dd7] [ERROR] Errors: 
[ae075ee89251cdf1536d0b3387c9c8f1ca750dd7] [ERROR]   SpamAssassinContract.deletingSpamMessageShouldNotImpactSpamAssassinState:557 » ConditionTimeout
```
test this please

File: server/container/cli-integration/src/test/java/org/apache/james/cli/DataCommandsIntegrationTest.java [hunk 5]
```diff
@@ -146,6 +148,8 @@ public void addAddressMappingShouldWork() throws Exception {
 
     @Test
     public void listMappingsShouldWork() throws Exception {
+        dataProbe.addDomain(DOMAIN);
```
Comment (rouazana) [inline server/container/cli-integration/src/test/java/org/apache/james/cli/DataCommandsIntegrationTest.java]: can you add some test to demonstrate what happens with the cli if the domain is missing?

**Comment by Arsnael** [comment]
Was :green_apple:  before squashing

**Comment by Arsnael** [comment]
```
[651044a1ea8784b44c39d40fd7d87777e06ef977] [ERROR] Errors: 
[651044a1ea8784b44c39d40fd7d87777e06ef977] [ERROR]   FilterTest.messageShouldBeAppendedInSpecificMailboxWhenFromRuleMatches:749 » ConditionTimeout
```
test this please

**Comment by Arsnael** [comment]
Was :green_apple:  , just restarted a build by mistake with an other task so I killed it

**Pull request merged**